### PR TITLE
Add back implementation for OpenLineage extractor for BigQueryInsertJobAsync Operator

### PR DESCRIPTION
**Pull request by pankajkoti**
This reverts commit 39102d01bc95fdb74635462ff3ee896b092699d8  to add back implementation 
for OpenLineage extractor for BigQueryInsertJobAsyncOperator. 
It was temporarily removed for release 1.3.0

Also, addressed Kaxil's comments on PR #290

### CODE REVIEW

**Review by bharanidharan14 — APPROVED** [review]

**Review by phanikumv — APPROVED** [review]

File: astronomer/providers/google/cloud/extractors/bigquery_async_extractor.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    def _get_big_query_client(self) -> Client:
+        """
+        Gets the BigQuery client to fetch job metadata.
+        The method checks whether a connection hook is available with the Airflow configuration for the operator, and
+        if yes, returns the same connection. Otherwise, returns the Client instance of``google.cloud.bigquery``.
+        """
+        hook = self.operator.hook
+        return hook.get_client(project_id=hook.project_id, location=hook.location)
+
+    def _get_xcom_bigquery_job_id(self, task_instance: TaskInstance) -> Any:
```
Comment (pankajastro) [inline astronomer/providers/google/cloud/extractors/bigquery_async_extractor.py:35]: Can we keep return type `str`?

Comment (pankajkoti) [reply astronomer/providers/google/cloud/extractors/bigquery_async_extractor.py:35]: The method return the ouput of `xcom_pull` method which declares the return type to `Any`. I had kept the return type to `str` as in our case the jobID will be string, but mypy complained that I have to keep it aligned with the method's return type and make it to `Any`

File: setup.cfg [hunk 2]
```diff
@@ -34,7 +34,7 @@ packages = find_namespace:
 include_package_data = true
 namespace_packages = astronomer,astronomer.providers
 install_requires =
-    apache-airflow>=2.2.0
+    apache-airflow>=2.3.0
```
Comment (kaxil) [inline setup.cfg:37]: Does this not work with Airflow 2.2.x @pankajkoti @phanikumv -- Bumping the pin means the providers won't be installable for users on 2.2.x which is what we don't want.

Comment (pankajkoti) [reply setup.cfg:37]: We observed an issue where some piece of code was calling xcom.execution_date which(the column) does not exist in the upgraded DB for 2.3.0 of our base image.

**Pull request merged**